compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3666,17 +3712,50 @@ OMR::Z::TreeEvaluator::ternaryEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-         bool is64BitRegister = (trueReg->getKind()==TR_GPR64 || cg->use64BitRegsOn32Bit());
-         generateRRFInstruction(cg, is64BitRegister ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg->getHighOrder(), falseReg->getHighOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
-         generateRRFInstruction(cg, is64BitRegister ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg->getLowOrder(), falseReg->getLowOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
-      }
+      // Load on condition is supported on z196 and up
+      if (cg->getS390ProcessorInfo()->supportsArch(TR_S390ProcessorInfo::TR_z196))
+         {
+         if (comp->getOption(TR_TraceCG))
+            {
+            traceMsg(comp, "Emmitting a LOCR with resulReg = %p falseReg->getRegister = %p\n",trueReg,falseReg->getRegister());
+            }
+
+         if (trueReg->getRegisterPair() || falseReg->getRegisterPair())
+            {
+            // Sanity check
+            TR_ASSERT(trueReg->getRegisterPair() && falseReg->getRegisterPair(), "Both false and true registers should be register pairs");
+
+            auto mnemonic = trueReg->getKind() == TR_GPR64 || cg->use64BitRegsOn32Bit() ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR;
+
+            generateRRFInstruction(cg, mnemonic, node, trueReg->getHighOrder(), falseReg->getHighOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC) >> 4, true);
+            generateRRFInstruction(cg, mnemonic, node, trueReg->getLowOrder(), falseReg->getLowOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC) >> 4, true);
+            }
+         else
+            {
+            auto mnemonic = trueVal->getOpCode().is8Byte() ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR;
+
+            generateRRFInstruction(cg, mnemonic, node, trueReg, falseReg->getRegister(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
+            }
+         }
       else
-         generateRRFInstruction(cg, trueVal->getOpCode().is8Byte() ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg, falseReg->getRegister(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
+         {
+         TR::LabelSymbol *branchDestination = TR::LabelSymbol::create(cg->trHeapMemory(),cg);
+         branchDestination->setEndInternalControlFlow();
+         TR::Instruction *brcInst = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BNERC,
+               node, branchDestination); //, cursor);
+         brcInst->setStartInternalControlFlow();
+         TR::Instruction *conditionalLoad = generateRRInstruction(cg, trueVal->getOpCode().is8Byte() ?
+               TR::InstOpCode::LGR : TR::InstOpCode::LR, node, trueReg, falseReg);
+         TR::RegisterDependencyConditions * conditions =
+               new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 4, cg);
```

Comment:
The z196 path has code to handle if trueReg/falseReg is a register pair.  However, the non-z196 path doesn't seem to handle this case.  Is that okay?